### ISSUE TITLE
Provide a way to specify the cafile when importing dashboards

### ISSUE THREAD

**Issue by ppf2** [elastic/beats#3121]
It will be helpful to provide a -cafile parameter so users can include the trusted cert chain when using import dashboards.  

```
./import_dashboards -es https://localhost:9200 -user elastic -pass changeme
fail to set the mapping. Error: Put https://localhost:9200/.kibana/_mapping/search: x509: certificate signed by unknown authority
Create temporary directory /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169
Downloading https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.0.zip
Unzip archive  /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169
/var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169/beats-dashboards-5.0.0/filebeat
Import directory  /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169/beats-dashboards-5.0.0/filebeat/index-pattern
Import index to /.kibana/index-pattern/filebeat-* from /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169/beats-dashboards-5.0.0/filebeat/index-pattern/filebeat.json
ERROR:  couldn't load json. Error: Put https://localhost:9200/.kibana/index-pattern/filebeat-*: x509: certificate signed by unknown authority
fail to import index-pattern: fail to load directory /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169/beats-dashboards-5.0.0/filebeat/index-pattern: error loading /var/folders/8j/3y6vjnyd5kz881f91q9dymr40000gn/T/tmp888949169/beats-dashboards-5.0.0/filebeat/index-pattern/filebeat.json: couldn't load json. Error: Put https://localhost:9200/.kibana/index-pattern/filebeat-*: x509: certificate signed by unknown authority
```

**Comment by brandonmensing** [comment]
Progress being made, please don't make a specific promise to the customer on ETA.

https://github.com/elastic/beats/pull/3139

**Comment by hanniballb** [comment]
Hi guys I am having exactly the same problem with the version 5.1.1. for packetbeat "beats-dashboards-5.1.1/packetbeat".

Anybody has an idea ??

**Comment by andrewkroh** [comment]
Fixed in #3139 + #3163.

@hanniballb You can use the `import_dashboard` from the [snapshot](https://beats-nightlies.s3.amazonaws.com/index.html?prefix=packetbeat/) build if you want to test out the changes. 

@elastic/beats Should we backport these to the 5.1 branch?

**Comment by hanniballb** [comment]
Providing more details...

I have mu ELKstack server on let's call it SERVER.1 and packetbeat SERVER.2.

I run  ./import_dashboards -es http://ip.ip.ip.ip:9200   (on SERVER.2) where my packetbeat sits
and I am getting this message.

Import search  /tmp/tmp444724439/beats-dashboards-5.1.1/packetbeat/search/Thrift-errors.json
fail to import dashboards: fail to load directory /tmp/tmp444724439/beats-dashboards-5.1.1/packetbeat/dashboard: error loading /tmp/tmp444724439/beats-dashboards-5.1.1/packetbeat/dashboard/Packetbeat-Cassandra.json: fail to read /tmp/tmp444724439/beats-dashboards-5.1.1/packetbeat/visualization/Cassandra:-ResponseKeyspace.json. Error: open /tmp/tmp444724439/beats-dashboards-5.1.1/packetbeat/visualization/Cassandra:-ResponseKeyspace.json: no such file or directory

**Comment by andrewkroh** [comment]
@hanniballb You could extract the linux/x86_64 tar.gz file and grab only the import_dashboard tool if you want to test the `-cacert` option. You will have to manually specify the URL to the dashboards that you want to install because by default it will be looking for the wrong version (e.g. use `-url https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip` for v5.0.2).

The Cassandra error is unrelated. See https://github.com/elastic/beats/issues/3140

**Comment by hanniballb** [comment]
@andrewkroh  the solution you have suggested worked for **packetbeat**. 

` ./import_dashboards -es http://my_elasticSearch_server:9200 -url https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip`


however it did not work for **filebeat**

 `./import_dashboards -es http://my_elasticSearch_server:9200 -url https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip`
Create temporary directory /tmp/tmp771405499
Downloading https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip
Unzip archive  /tmp/tmp771405499
/tmp/tmp771405499/beats-dashboards-5.0.2/filebeat
Import directory  /tmp/tmp771405499/beats-dashboards-5.0.2/filebeat/index-pattern
Import index to /.kibana/index-pattern/filebeat-* from /tmp/tmp771405499/beats-dashboards-5.0.2/filebeat/index-pattern/filebeat.json
No directory /tmp/tmp771405499/beats-dashboards-5.0.2/filebeat/dashboard
/tmp/tmp771405499/beats-dashboards-5.0.2/metricbeat
/tmp/tmp771405499/beats-dashboards-5.0.2/packetbeat
/tmp/tmp771405499/beats-dashboards-5.0.2/winlogbeat

**Comment by andrewkroh** [comment]
@hanniballb  That's normal. There are no Filebeat dashboards. There's only and index-pattern.

**Comment by hanniballb** [comment]
@andrewkroh  Thank you very much for consulting, I am kinda new in ELK stack.

**Comment by vaibhavjpr** [comment]
Hi @andrewkroh 

any idea on this..

[root@slc10fha kibana]# cd ../filebeat/scripts/
[root@slc10fha scripts]# ./import_dashboards -es http://localhost:9200 -url https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip
fail to set the mapping. Error: 401 Unauthorized
Create temporary directory /tmp/tmp265329241
Downloading https://artifacts.elastic.co/downloads/beats/beats-dashboards/beats-dashboards-5.0.2.zip
Unzip archive  /tmp/tmp265329241
/tmp/tmp265329241/beats-dashboards-5.0.2/filebeat
Import directory  /tmp/tmp265329241/beats-dashboards-5.0.2/filebeat/index-pattern
Import index to /.kibana/index-pattern/filebeat-* from /tmp/tmp265329241/beats-dashboards-5.0.2/filebeat/index-pattern/filebeat.json
ERROR:  couldn't load json. Error: 401 Unauthorized
fail to import index-pattern: fail to load directory /tmp/tmp265329241/beats-dashboards-5.0.2/filebeat/index-pattern: error loading /tmp/tmp265329241/beats-dashboards-5.0.2/filebeat/index-pattern/filebeat.json: couldn't load json. Error: 401 Unauthorized

[root@slc10fha scripts]#

**Comment by andrewkroh** [comment]
@vaibhavjpr  Use https://discuss.elastic.co/c/beats for questions.

**Comment by andrewkroh** [comment]
Closing. This is fixed in v5.2.0 with new options for `-cafile`, `-cert`, `-key`, and `-insecure`.
